### ISSUE TITLE
Multiple issues on NeoVim

### ISSUE THREAD

**Issue by Ron89** [jceb/vim-orgmode#213]
Multiple functionalities is not working on Neovim:
* New Headings Above/Below-before-content not working
* New Heading Below not working on last Heading
* New checkbox below not working on last checkbox

**Comment by Ron89** [comment]
As reviewing the code, I think at least the second and third issue listed above are from NeoVim's ["Out-of-Bound" handling](https://github.com/neovim/neovim/issues/4058). Currently the issue is still open and have a [pull request](https://github.com/neovim/python-client/issues/155) not implemented yet. 

I also have written [some fix](https://github.com/Ron89/vim-orgmode/tree/neovim_support) to bypass the issue, essentially by revising `VimBuffer:write` and `EditCheckbox:new_checkbox`, so that `Vim.current.buffer` first append a newline. But because the heading's handling is a bit tricky, I haven't locate the part responsible for the first issue as yet.

**Comment by jceb** [comment]
Thanks for figuring out the differences and creating workarounds.  Much appreciated!  Let me know when there is something to be merged.

**Comment by Ron89** [comment]
Thanks to your comment on issue #196, I realized that problem with `New
Headings Above/Below-before-content not working` is actually a separate issue(I
didn't realize it because I run vim in GUI and neovim on CLI as it don't have
a stable GUI yet). I checked the leader key mapping for those functions and
they work fine. 

I just sent in the pull request #214. It's a pretty ugly workaround, but works
on my vim and neovim.
